core/src/lib.rs

```diff
@@ -329,6 +329,91 @@ impl From<f64> for TypedValue {
     }
 }
 
+pub trait TypedValueConverter {
+    fn as_entid(self) -> Option<Entid>;
```

Comment:
All of these are `into`s, not `as`. I think `as` or `to` are what we want here:

```
fn to_entid(&self) -> Option<Entid>;
fn as_kw(&self) -> Option<&NamespacedKeyword>;
```

Reply:
Further: what you've defined here is effectively `TryFrom` (and see `IntoThing` elsewhere in Mentat). You might consider whether you really want a generic trait here: `AsOption<T>`, perhaps?